charts/netdata/1.0.10/migrations/migrate

```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/python3
+import json
+import os
+import sys
+import subprocess
+from pathlib import Path
+
+from middlewared.client import Client
+from middlewared.service import ValidationErrors, CallError
+
+
+def path_in_locked_datasets(path: str) -> bool:
+    with Client() as c:
+        return c.call('pool.dataset.path_in_locked_datasets', path)
+
+
+def get_configured_user_group(path: str) -> dict:
+    with Client() as c:
+        return c.call('filesystem.stat', path)
+
+
+def get_host_path_attachments(path: str) -> set:
+    with Client() as c:
+        return {
+            attachment['type']
+            for attachment in c.call('pool.dataset.attachments_with_path', path)
+            if attachment['type'].lower() not in ['kubernetes', 'chart releases']
+        }
+
+
+def get_kubernetes_config() -> dict:
+    with Client() as c:
+        return c.call('kubernetes.config')
+
+
+def validate_host_path(path: str, schema_name: str, verrors: ValidationErrors) -> None:
+    """
+    These validations are taken from `FilesystemService._common_perm_path_validate`.
+    """
+    schema_name += ".migration.chown"
+    p = Path(path)
+    if not p.is_absolute():
+        verrors.add(schema_name, f"Must be an absolute path: {path}")
+
+    if p.is_file():
+        verrors.add(schema_name, f"Recursive operations on a file are invalid: {path}")
+
+    if not p.absolute().as_posix().startswith("/mnt/"):
+        verrors.add(
+            schema_name,
+            f"Changes to permissions on paths that are not beneath the directory /mnt are not permitted: {path}"
+        )
+    elif len(p.resolve().parents) == 2:
+        verrors.add(schema_name, f"The specified path is a ZFS pool mountpoint: {path}")
+
+    # Make sure that dataset is not locked
+    if path_in_locked_datasets(path):
+        verrors.add(schema_name, f"Dataset is locked at path: {path}.")
+
+    # Validate attachments
+    if attachments := get_host_path_attachments(path):
+        verrors.add(schema_name, f"The path '{path}' is already attached to service(s): {', '.join(attachments)}.")
+
+
+def validate_path(uid, gid, host_path, release_name):
+    verrors = ValidationErrors()
+
+    current_config = get_configured_user_group(host_path)
+    if current_config["uid"] == uid and current_config["gid"] == gid:
+        return
+
+    validate_host_path(host_path, release_name, verrors)
+    verrors.check()
+
+
+def chown_path(uid, gid, host_path):
+    acltool = subprocess.run([
+        "/usr/bin/nfs4xdr_winacl",
+        "-a", "chown",
+        "-O", str(uid), "-G", str(gid),
+        "-r",
+        "-c", host_path,
+        "-p", host_path], check=False, capture_output=True
+    )
```

Comment:
Question:
Does this support: Non-ACL, POSIX-ACL *and* NFSv4 ACL's?

Reply:
@sonicaj Maybe you can related, does this support all three possible permissions variants?

Reply:
Verified to be the case.
Resolved.